### ISSUE TITLE
Fullstack Stacks Development Guide

### ISSUE THREAD

**Issue by will-corcoran** [stacksgov/critical-bounties#26]
Recipient: Leopadel
Bounty: $2,400

### Summary

- **Objective**: Create a comprehensive guide for full stack development on the Stacks blockchain.
- **Need**: Identified lack of documentation and guides for building full stack applications on Stacks.
- **Proposal**: Develop "The Complete Guide to Full Stack Stacks Development" with step-by-step tutorials.

### Guide Content

1. **Clarity Contract + Testing**:
    - Create a simple contract.
    - Write tests using Clarinet.
2. **Next.js App with Authentication + Contract Interaction**:
    - Develop a Next.js app where users connect their wallet.
    - Interact with the contract using stacks.js.
3. **Chainhook Indexer with Database (SQLite)**:
    - Create an indexer to listen to contract events.
    - Store data in a SQLite database.
    - Display data from the database in the frontend.

### Reference

- The content will be similar in structure to the full stack Ethereum development guide on freeCodeCamp.

### Distribution

- Platforms like Sigle, Dev.to, and Hashnode will be considered for broader reach.

## Deliverables

- **Deliverable 1**: Comprehensive guide on Clarity contract development and testing.
- **Deliverable 2**: Tutorial for building a Next.js app with authentication and contract interaction using stacks.js.
- **Deliverable 3**: Guide on creating a Chainhook indexer with SQLite database and integrating it with a frontend.

## Roadmap & Timeline

- **Phase 1**: Research and content planning (2 weeks).
    - Identify key topics and structure for the guide.
    - Plan the step-by-step tutorials and examples.
- **Phase 2**: Content creation (8 weeks).
    - Develop and document Clarity contract and testing guide.
    - Create the Next.js app tutorial with wallet integration and contract interaction.
    - Write the guide for Chainhook indexer and database setup.
- **Phase 3**: Review and editing (2 weeks).
    - Peer review and edit the content for accuracy and clarity.
- **Phase 4**: Distribution and promotion (4 weeks).
    - Publish the guide on selected platforms.
    - Promote through social media and developer communities.

**Comment by Faizalhidayat** [comment]
this task still open ?

**Comment by will-corcoran** [comment]
Hi @pradel - 

I hope you are well.  Now that we are about a month into having awarded this Critical Bounty, we would kindly request that you provide an update here for the community.  Before the end of the week (Friday, August 16th) can you please provide us an update on the status of your Critical Bounty, please make sure to include:
- Summary of CB-related work that has been done to date
- Summary of CB roadmap from here to completion
- Flag an potential blockers
- Anticipated date of completion
- Link to all related repos (remember all funded work should be open source)

Thanks!
Will

cc: @shaktistacks

**Comment by pradel** [comment]
Hi Will,

Due to the work on Stackspulse being delayed, I just started this week on the content planning. I expect to have Milestone 1 ready in around 2 weeks so I can share that part once it's ready.
For phase 3, do you know some people who can help at the foundation or others?
For phase 4, who should I contact to plan the release schedule and see what is the best way to release it to reach a wider audience?

I don't expect any blockers.

**Comment by will-corcoran** [comment]
Hi @pradel 

 We're exploring allocating our remaining 2024 grants budget and considering follow-up funding for current Critical Bounty recipients. If you have ideas to enhance your previously awarded work, we'd like to hear from you.

### **Next Steps:**

1. If you're interested in follow-up funding, please let us know by the end of this week.
2. You can either:
  -   Post your proposal as a comment on this issue (preferred for final submissions)
  -   Email me your initial thoughts if you're still developing your idea or prefer privacy

### **Proposal Details:**

- Please include:
  - Additional scope or enhancements
  - Associated costs
  - Justification for how a follow-up grant could significantly boost the ROI of the initial grant

We look forward to your ideas!

Best,
Will

**Comment by pradel** [comment]
@will-corcoran Update:

I've completed the first two parts of the content. The first focuses on Clarity Contracts and Testing. The second part focuses on Next.js App with Authentication and Contract Interaction.

You can find a temporary link [here](https://leopradel-git-feat-full-stack-stacks-guide-leopradels-projects.vercel.app/blog/full-stack-guide-stacks)

I will be ready for the next phase after publishing part 3, which I expect to finish in two weeks.

**Comment by pradel** [comment]
@will-corcoran update:

- the chainhook guide is actually quite big so I decided to split it into another separate article
- I am having some issues getting chainhooks working on testnet and this is currently blocking me. Once I unblock the chainhook issues I am having I need probably a few days to finish the content and we should be good to go to phase 3.

**Comment by will-corcoran** [comment]
excellent.  thanks for the update @pradel .  Please let me know if there is anyone I can help connect you with to get you unblocked.  Best, Will

**Comment by will-corcoran** [comment]
@pradel is this completed yet?
